diff --git a/node_modules/drizzle-orm/op-sqlite/session.js b/node_modules/drizzle-orm/op-sqlite/session.js
index 4b644d9..95fe753 100644
--- a/node_modules/drizzle-orm/op-sqlite/session.js
+++ b/node_modules/drizzle-orm/op-sqlite/session.js
@@ -85,7 +85,8 @@ class OPSQLitePreparedQuery extends SQLitePreparedQuery {
       const params = fillPlaceholders(query.params, placeholderValues ?? {});
       logger.logQuery(query.sql, params);
       return await this.queryWithCache(query.sql, params, async () => {
-        return client.execute(query.sql, params).rows?._array || [];
+        const result = await client.execute(query.sql, params);
+        return result.rows ?? [];
       });
     }
     const rows = await this.values(placeholderValues);
@@ -100,7 +101,8 @@ class OPSQLitePreparedQuery extends SQLitePreparedQuery {
     logger.logQuery(query.sql, params);
     if (!fields && !customResultMapper) {
       const rows2 = await this.queryWithCache(query.sql, params, async () => {
-        return client.execute(query.sql, params).rows?._array || [];
+        const result = await client.execute(query.sql, params);
+        return result.rows ?? [];
       });
       return rows2[0];
     }
@@ -118,7 +120,8 @@ class OPSQLitePreparedQuery extends SQLitePreparedQuery {
     const params = fillPlaceholders(this.query.params, placeholderValues ?? {});
     this.logger.logQuery(this.query.sql, params);
     return await this.queryWithCache(this.query.sql, params, async () => {
-      return await this.client.executeRawAsync(this.query.sql, params);
+      const result = await this.client.executeRawAsync(this.query.sql, params);
+      return result.rawRows ?? result;
     });
   }
   /** @internal */
